src/models.js

```diff
@@ -18,35 +17,53 @@ export class Point {
     return new Point(this.row +1, this.col)
   }
 
+  isAbove( p2 ) {
+    return this.row === p2.row +1 && this.col === p2.col
+  }
+
   sameAs( p2 ) {
     return this.row === p2.row && this.col === p2.col
   }
-
 }
 
 export class Shape {
   constructor( name, rotator ) {
     this.name = name
     this.rotator = rotator
     this.classString = `square ${name}`
-    this.isPaused = false
   }
   pointsRotated( rotation ) {
     return this.rotator( rotation )
   }
+}
+
+const ROTATIONS = ['N', 'E', 'S', 'W']
+
+class Rotation {
+  constructor( current ) {
+    this.index = ROTATIONS.indexOf( current )
+  }
 
+  next() {
+    return ROTATIONS[ ( this.index + 1 ) % 4 ]
+  }
+
+  previous() {
+    return ROTATIONS[ ( this.index - 1 ) % 4 ]
+  }
 }
 
 export class Piece {
-  constructor(shape, offset = new Point (1, 10) ) {
+  constructor(shape, offset = new Point (-2, 5) ) {
     this.shape = shape
     this.offset = offset
     this.rotation = 'N'
     this.classString = shape.classString
   }
 
   points() {
-    return this.shape.pointsRotated( this.rotation ).map( ( point, ix ) => point.add( this.offset ) );
+    return this.shape.pointsRotated( this.rotation )
```

Comment:
This feels like it may be another violation of the Law of Demeter (reaching through shape into another thing to perform an operation)...